function-logs/handler.go

```diff
@@ -0,0 +1,99 @@
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openfaas/openfaas-cloud/sdk"
+)
+
+// Handle grabs the logs for the fns that are named in the input
+func Handle(req []byte) string {
+
+	user := string(req)
+	var function string
+	if len(user) == 0 {
+		if query, exists := os.LookupEnv("Http_Query"); exists {
+			vals, _ := url.ParseQuery(query)
+			userQuery := vals.Get("user")
+			function = vals.Get("function")
+			if len(userQuery) > 0 {
+				user = userQuery
+			}
+		}
+	}
+
+	if len(user) == 0 {
+		return "User is required as POST or querystring i.e. ?user=alexellis."
+	}
+
+	c := http.Client{
+		Timeout: time.Second * 3,
+	}
+
+	gatewayURL := os.Getenv("gateway_url")
+
+	httpReq, _ := http.NewRequest(http.MethodGet, gatewayURL+"system/logs", nil)
+
+	query := url.Values{}
+	query.Add("name", function)
+	query.Add("follow", "false")
+	query.Add("verify_label", fmt.Sprintf("com.openfaas.cloud.git-owner=%s", user))
+
+	addAuthErr := sdk.AddBasicAuth(httpReq)
+	if addAuthErr != nil {
+		log.Printf("Basic auth error %s", addAuthErr)
```

Comment:
Seems like the end of the invocation if the basic auth failed?

Reply:
changed to return an error